sdk/clientcore/core/src/main/java/io/clientcore/core/http/models/AuthMetadata.java

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package io.clientcore.core.http.models;
+
+import io.clientcore.core.credentials.oauth.OAuthTokenRequestContext;
+
+import java.util.List;
+
+/**
+ * Auth Metadata represents the authentication related data provided at Http Request level to the pipeline policies.
+ */
+public final class AuthMetadata {
+    private List<AuthScheme> authScheme;
```

Comment:
nit: this should be `authSchemes` as well but can be done in a separate PR

Reply:
To update in follow up PR.